sc/social/like/browser/viewlets.py

```diff
@@ -57,15 +65,33 @@ def update(self):
         return
 
 
-class SocialMetadataViewlet(BaseLikeViewlet):
-    """Viewlet used to insert metadata into page header
+class SocialMetadataViewlet(ViewletBase):
+    """Viewlet used to insert social metadata into page header
+    including Open Graph protocol, Facebook and Twitter properties.
     """
-    render = ViewPageTemplateFile('templates/metadata.pt')
-    render_method = 'metadata'
+
+    index = ViewPageTemplateFile('templates/metadata.pt')
+
+    def update(self):
+        registry = getUtility(IRegistry)
```

Comment:
Any reason for not using plone.api directly since it's already a dependency of sc.social.like https://github.com/collective/sc.social.like/blob/40cfc4c09b08892837411561efe9447870c297ce/setup.py#L45 in `app_id`, `admins` and `twitter_site`?

Reply:
because plone.api sucks on handling the registry as I would have to add one call to `get_registry_record()` for every field and that leads to more lines of code at the end.

we need to implement something like `api.registry.get_record()` and `api.registry.get_proxy()`.